src/Pages/ArticlePage.jsx

```diff
@@ -1,9 +1,64 @@
-const ArticlePage = () => {
-    return (
-        <section>
-            <p>ArticlePage</p>
-        </section>
-    )
-}
-
-export default ArticlePage
\ No newline at end of file
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { getArticleById } from "../apis/api";
+import { timeDiffToCurrentDate } from "../utils/utils";
+import Footer from "../layouts/Footer";
+
+const ArticlePage = ({ loaded, setLoaded }) => {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState({});
+
+  useEffect(() => {
+    const controller = new AbortController();
+
+    const fetchArticle = async () => {
+      setLoaded(false);
+
+      try {
+        const response = await getArticleById(article_id, controller.signal);
+        setArticle(response);
+      } catch (error) {
+        console.log("error at api request", error);
+        //noop
+      } finally {
+        setLoaded(true);
+      }
+    };
+    fetchArticle();
+    return () => {
+      controller.abort();
+    };
+  }, []);
+
+  if (!loaded) return <p>Loading</p>;
```

Comment:
currently your loading state isn't kicking in for me when i throttle my internet. I want you to try it out but it completely bypasses this if statement by the looks of things and tries to render empty information.